### PR TITLE
Add information about beach-head services

### LINKED ISSUE #780: Write documentation for service templates in official docs

### PR DESCRIPTION
**Pull request by mestadler**
Initial PR for the servicetemplate usage guide.

### CODE REVIEW

**Comment by DinaBelova** [comment]
@mestadler can you please 
1) share what's the status of this PR? Afair this was not in final stage yet, we discussed it with you and @wahabmk  - are there any changes since that time? I do not see them in the commits, so maybe it was not yet updates, correct?
2) as a suggestion - can you also please extend quick start guide section in a form of separate commit in PR? Will be nice to have section w/ kyverno or nginx installation in quickstart

**Comment by wahabmk** [comment]
I re-organized some of the information already existing in this PR + more info into the following:
1. `deploy-services-managedcluster.md`: Shows how beach-head services can be deployed with a `ManagedCluster` object.
2. `create-multiclusterservice.md`: Shows how beach-head services can be deployed with a `MultiClusterService` object. Some portions are shared with ManagedCluster, so those portions refer to the `deploy-services-managedcluster.md` page.
3. Some additional information about `ServiceTemplate` in other sections of the docs as appropriate.

I think this is a good start for now for information on beach-head services.

**Comment by wahabmk** [comment]
@jjainschigg-r Feel free to add Nick Chase as well, I couldn't find his github handle.

File: docs/quick-start/2a-installation.md [hunk 1]
```diff
@@ -59,16 +59,58 @@ k0smotron-controller-manager-infrastructure-7f77f55675-tv8vb   2/2     Running
...
+
 If you have fewer pods, give 2A more time to reconcile all the pods.
 
-As a second verification, check that the example `ClusterTemplate` objects have
-been installed and are valid:
+### Verify 2A templates have been successfully reconciled
+
+For additional verification, check that the example templates packaged with 2A have
+been installed and are valid.
+
```
Comment (BROngineer) [inline docs/quick-start/2a-installation.md:87]: I'd describe the naming convention for templates we use: `template-name-sem-ver-with-dashes`, replace concrete versions in provided outputs by X-Y-Z, like `cluster-api-X-Y-Z`, add separate pages for predefined templates:

- `./pre-defined-provider-templates.md`
- `./pre-defined-cluster-templates.md`
- `./pre-defined-service-templates.md` (#66)

where actual versions will be listed. And add references to them, like:

> The output should be like the following:
>
> ```sh
> NAME                                 VALID
> cluster-api-X-Y-Z                    true
> cluster-api-provider-aws-X-Y-Z       true
> cluster-api-provider-azure-X-Y-Z     true
> cluster-api-provider-vsphere-X-Y-Z   true
> hmc-X-Y-Z                            true
> k0smotron-X-Y-Z                      true
> projectsveltos-X-Y-Z                 true
> ```
>
> The actual list of pre-defined templates with versions could be found {link to corresponding file}

I believe it will simplify keeping documentation accurate in the future (for instance cluster template versions example is already outdated).

File: docs/usage/create-multiclusterservice.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+1. ManagedCluster `dev-cluster-1` manages deployment of kyverno (v3.2.6) and ingress-nginx (v4.11.0) with `servicesPriority=100` on its cluster.
+2. ManagedCluster `dev-cluster-2` manages deployment of ingress-nginx (v4.11.0) with `servicesPriority=500` on its cluster.
+3. MultiClusterService `global-ingress` manages deployment of ingress-nginx (v4.11.3) with `servicesPriority=300` on both clusters.
+
+This scenario presents a conflict on both the clusters as the MultiClusterService is attempting to deploy v4.11.3 of ingress-nginx
+on both whereas the ManagedCluster for each is attempting to deploy v4.11.0 of ingress-nginx.
+
+This is where `.spec.servicesPriority` can be used to specify who gets the priority. Higher number means higer priority and vice versa. In this example:
+1. ManagedCluster `dev-cluster-1` would lose and its ingress-nginx (v4.11.3) would be deployed on its cluster.
+2. ManagedCluster `dev-cluster-2` would win and ingress-nginx (v4.11.0) would be deployed on its cluster.
```
Comment (BROngineer) [inline docs/usage/create-multiclusterservice.md]: I'd reword it as:

> 1. `MultiClusterService` "global-ingress" will take precedence over `ManagedCluster` "dev-cluster-1" and ingress-nginx (v4.11.3) defined in `MultiClusterService` object will be deployed on the cluster.
> 2. `ManagedCluster` "dev-cluster-2" will take precedence over `MultiClusterService` "global-ingress" and ingress-nginx (v4.11.0) defined in `ManagedCluster` object will be deployed on the cluster.

File: docs/usage/deploy-services-managedcluster.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+      namespace: motel
+      template: motel-regional-0-1-1
+      values: |
+        victoriametrics:
+          vmauth:
+            ingress:
+              host: vmauth.hmc0.example.net
+            credentials:
+              username: motel
+              password: motel
```
Comment (BROngineer) [inline docs/usage/deploy-services-managedcluster.md:131]: Does not related to documentation itself, but need to be fixed asap. Issue: https://github.com/Mirantis/hmc/issues/817

**Review by BROngineer — DISMISSED** [review]
lgtm

**Review by DinaBelova — APPROVED** [review]
@mestadler I think we can land this one, and in case if there will be any more suggestions around how to improve service templates docs, we'll go from there

**Review by mestadler — COMMENTED** [review]
Ack
